src/Http/Client.php

```diff
@@ -0,0 +1,186 @@
+
+        return $this->execute($request);
+    }
+
+    /**
+     * @param string $path
+     * @param null   $body
+     * @param array  $query
+     *
+     * @return mixed
+     *
+     * @throws ClientExceptionInterface
+     * @throws HTTPRequestException
+     */
+    public function post($path, $body = null, $query = [])
+    {
+        $request = $this->requestFactory->createRequest(
+            'POST',
+            $this->baseUrl.$path.$this->buildQueryString($query)
+        )->withBody($this->streamFactory->createStream(json_encode($body)));
+
+        return $this->execute($request);
+    }
+
+    public function put($path, $body = null, $query = [])
+    {
+        $request = $this->requestFactory->createRequest(
+            'PUT',
+            $this->baseUrl.$path.$this->buildQueryString($query)
+        )->withBody($this->streamFactory->createStream(json_encode($body)));
+
+        return $this->execute($request);
+    }
+
+    /**
+     * @param string $path
+     * @param null   $body
+     * @param array  $query
+     *
+     * @return mixed
+     *
+     * @throws ClientExceptionInterface
+     * @throws HTTPRequestException
+     */
+    public function patch($path, $body = null, $query = [])
+    {
+        $request = $this->requestFactory->createRequest(
+            'PATCH',
+            $this->baseUrl.$path.$this->buildQueryString($query)
+        )->withBody($this->streamFactory->createStream(json_encode($body)));
+
+        return $this->execute($request);
+    }
+
+    /**
+     * @param $path
+     * @param array $query
+     *
+     * @return mixed
+     *
+     * @throws ClientExceptionInterface
+     * @throws HTTPRequestException
+     */
+    public function delete($path, $query = [])
+    {
+        $request = $this->requestFactory->createRequest(
+            'DELETE',
+            $this->baseUrl.$path.$this->buildQueryString($query)
+        );
+
+        return $this->execute($request);
+    }
+
+    /**
+     * @return mixed
+     *
+     * @throws ClientExceptionInterface
+     * @throws HTTPRequestException
+     */
+    private function execute(RequestInterface $request)
+    {
+        foreach ($this->headers as $header => $value) {
+            $request = $request->withAddedHeader($header, $value);
+        }
+
+        return $this->parseResponse($this->http->sendRequest($request));
+    }
+
+    private function buildQueryString(array $queryParams = []): string
+    {
+        return $queryParams ? '?'.http_build_query($queryParams) : '';
+    }
+
+    /**
+     * @return mixed
+     *
+     * @throws HTTPRequestException@
+     */
+    private function parseResponse(ResponseInterface $response)
+    {
+        if ($response->getStatusCode() >= 300) {
+            $body = json_decode($response->getBody()->getContents(), true);
+            throw new HTTPRequestException($response->getStatusCode(), $body);
+        }
+
+        return json_decode($response->getBody()->getContents(), true);
```

Comment:
Actually, it return an array so I think that we can typehint each method that use this method as a return, any thought? 🤔

Reply:
same as here. https://github.com/meilisearch/meilisearch-php/pull/43#discussion_r446290517

Reply:
Okay for me 🙂